crates/syntax/rust.ungram

```diff
@@ -614,7 +614,7 @@ TypeBoundList =
 
 TypeBound =
   Lifetime
-| ('?' | '~' 'const')? Type
+| ('~' 'const' | 'const')? 'async'? '?'? Type
```

Comment:
opted for this because it's a bit more accurate representation of the real grammar rustc parses. I can change it if desired, though.

Reply:
No that's fine as long as the possible tokens are represented.